### ISSUE TITLE
how to set proxy 

### ISSUE THREAD

**Issue by wangshuai-wuhan** [langgenius/dify#3079]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### Provide a description of the help you need

i want to set proxy for openai ,how can i do

**Comment by leslie2046** [comment]
@wangshuai-wuhan just add 3 lines in your docker-compose.yaml or like below:
![图片](https://github.com/langgenius/dify/assets/10449699/143ce460-b276-4771-b79c-776bebaeac8a)

**Comment by xyb** [comment]
> @wangshuai-wuhan just add 3 lines in your docker-compose.yaml or like below: ![图片](https://private-user-images.githubusercontent.com/10449699/318702637-143ce460-b276-4771-b79c-776bebaeac8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NTcxOTIsIm5iZiI6MTcyNTk1Njg5MiwicGF0aCI6Ii8xMDQ0OTY5OS8zMTg3MDI2MzctMTQzY2U0NjAtYjI3Ni00NzcxLWI3OWMtNzc2YmViYWVhYzhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDA4MjgxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxMTFlY2VkOWE2NzEyMDM1Yzg0ZmY5MzFhMzU2ZmE5MTUyY2Q4ZGVhMzFhNzQ2N2I5ZGVmY2MxZDk1YTRmMTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aHyCTWNvm1HAaE-H3mn77Iz7R5jOBdUGFu8-pk59z7g)

Add `sandbox` to `NO_PROXY` to avoid this error:
```
HTTP Request: POST http://sandbox:8194/v1/sandbox/run "HTTP/1.1 502 Bad Gateway
```

**Comment by nickhelion** [comment]
> > [@wangshuai-wuhan](https://github.com/wangshuai-wuhan) just add 3 lines in your docker-compose.yaml or like below: ![图片](https://private-user-images.githubusercontent.com/10449699/318702637-143ce460-b276-4771-b79c-776bebaeac8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NTcxOTIsIm5iZiI6MTcyNTk1Njg5MiwicGF0aCI6Ii8xMDQ0OTY5OS8zMTg3MDI2MzctMTQzY2U0NjAtYjI3Ni00NzcxLWI3OWMtNzc2YmViYWVhYzhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDA4MjgxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxMTFlY2VkOWE2NzEyMDM1Yzg0ZmY5MzFhMzU2ZmE5MTUyY2Q4ZGVhMzFhNzQ2N2I5ZGVmY2MxZDk1YTRmMTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aHyCTWNvm1HAaE-H3mn77Iz7R5jOBdUGFu8-pk59z7g)
> 
> Add `sandbox` to `NO_PROXY` to avoid this error:
> 
> ```
> HTTP Request: POST http://sandbox:8194/v1/sandbox/run "HTTP/1.1 502 Bad Gateway
> ```

For any one who see this and decide to adopt the method above, you have to add more hostname in NO_PROXY list, such as `plugin_daemon`. It was introduced in recent versions

**Comment by rty813** [comment]
@leslie2046 I set up an HTTP proxy for the API service in the Compose file. When I logged into the container, I saw that the env was working. 

<img width="405" height="215" alt="Image" src="https://github.com/user-attachments/assets/34c8367a-dcfa-4ffc-bdbf-73199aa06a7d" />

I also saw access logs on the proxy server when I accessed the target domain using curl. 
However, when I started a conversation using the Dify webpage, I couldn't see any access logs on the proxy server, suggesting that the http proxy wasn't working. What could be the reason?
